prov/sm2/src/sm2.h

```diff
@@ -176,11 +176,11 @@ static inline int sm2_match_tag(uint64_t tag, uint64_t ignore,
 static inline void sm2_generic_format(struct sm2_xfer_entry *xfer_entry,
```

Comment:
this is wild but the osu_alltoallw degredation seems to come from this patch.

Reply:
I would ignore this for now.

Reply:
I don't understand why the `sm2_generic_format` function should modify op_flags, and it doesn't seem to here or in your final commit.

Reply:
That is in `sm2_do_cma`.  In `sm2_generic_format` the op_flag of the caller should not be modified, but passing a pointer to the int suggests it can be.

I will propose a separate follow-up later, I don't think this small nit needs to hold up this merge.

Reply:
Fixed